stdlib/public/core/String.swift

```diff
@@ -925,91 +925,57 @@ extension String {
-    if let me = utf16Index.samePosition(
-      in: characters.unicodeScalars
-    )?.samePosition(in: characters) {
-      self = me
-    }
-    else {
-      return nil
-    }
-  }
-
-  /// Construct the position in `characters` that corresponds exactly to
-  /// `utf8Index`. If no such position exists, the result is `nil`.
-  ///
-  /// - Precondition: `utf8Index` is an element of
-  ///   `characters.utf8.indices`.
-  public init?(
-    _ utf8Index: String.UTF8Index,
-    within characters: String
-  ) {
-    if let me = utf8Index.samePosition(
-      in: characters.unicodeScalars
-    )?.samePosition(in: characters) {
-      self = me
-    }
-    else {
-      return nil
-    }
-  }
-
-  /// Returns the position in `utf8` that corresponds exactly
-  /// to `self`.
-  ///
-  /// - Precondition: `self` is an element of `String(utf8).indices`.
+extension String {
+  // TODO: swift-3-indexing-model - add docs
   @warn_unused_result
-  public func samePosition(
-    in utf8: String.UTF8View
-  ) -> String.UTF8View.Index {
-    return String.UTF8View.Index(self, within: utf8)
+  public func convert(
+    index index: String.UTF8Index,
+    from view: String.UTF8View
+  ) -> String.Index? {
+    fatalError("FIXME: swift-3-indexing-model: implement")
+
+    // FIXME: swift-3-indexing-model: rework the following
+//    if let me = utf8Index.samePosition(
+//      in: characters.unicodeScalars
+//      )?.samePosition(in: characters) {
+//        self = me
+//    }
+//    else {
+//      return nil
+//    }
   }
 
-  /// Returns the position in `utf16` that corresponds exactly
-  /// to `self`.
-  ///
-  /// - Precondition: `self` is an element of `String(utf16).indices`.
+  // TODO: swift-3-indexing-model - add docs
   @warn_unused_result
-  public func samePosition(
-    in utf16: String.UTF16View
-  ) -> String.UTF16View.Index {
-    return String.UTF16View.Index(self, within: utf16)
+  public func convert(
+    index index: String.UTF16Index,
+    from view: String.UTF16View
+  ) -> String.Index? {
+    fatalError("FIXME: swift-3-indexing-model: implement")
+
+    // FIXME: swift-3-indexing-model: rework the following
+//    if let me = utf16Index.samePosition(
+//      in: characters.unicodeScalars
+//      )?.samePosition(in: characters) {
+//        self = me
+//    }
+//    else {
+//      return nil
+//    }
   }
 
-  /// Returns the position in `unicodeScalars` that corresponds exactly
-  /// to `self`.
-  ///
-  /// - Precondition: `self` is an element of `String(unicodeScalars).indices`.
+  // TODO: swift-3-indexing-model - add docs
   @warn_unused_result
-  public func samePosition(
-    in unicodeScalars: String.UnicodeScalarView
-  ) -> String.UnicodeScalarView.Index {
-    return String.UnicodeScalarView.Index(self, within: unicodeScalars)
+  public func convert(
+    index index: String.UnicodeScalarIndex,
+    from view: String.UnicodeScalarView
+  ) -> String.Index? {
+    // FIXME: swift-3-indexing-model: verify cores match?
```

Comment:
StringCore does not need to match.  The prefixes should be identical though, but that's too expensive to check :(